dietpi/dietpi-software

```diff
@@ -5519,6 +5529,36 @@ Package: wireguard wireguard-dkms wireguard-tools\nPin: release n=bullseye\nPin-
 			G_EXEC cd /tmp/$G_PROGRAM_NAME
 
 		fi
+		
+		software_id=183 # Bitwarden RS
+                if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+                    Banner_Installing
+
+                    # Dependencies: https://github.com/dani-garcia/bitwarden_rs/wiki/Building-binary#dependencies
+                    DEPS_LIST='pkg-config libssl-dev'
+
+                    # Rust from https://rustup.rs/
+                    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain none
+                    source "/root/.cargo/env"
+
+                    # Download & Install
+                    INSTALL_URL_ADDRESS='https://api.github.com/repos/dani-garcia/bitwarden_rs/releases/latest'
+                    G_CHECK_URL "$INSTALL_URL_ADDRESS"
+                    local version=$(curl -sf "$INSTALL_URL_ADDRESS" | grep -m1 '^[[:blank:]]*"tag_name":' | cut -d \" -f 4)
+                    Download_Install "https://github.com/dani-garcia/bitwarden_rs/archive/$version.tar.gz"
+
+                    [[ -d '/opt/bitwarden_rs' ]] && G_EXEC rm -R /opt/bitwarden_rs
+                    G_EXEC mv bitwarden_rs-* /opt/bitwarden_rs
+
+                    # Build
+                    G_EXEC cd /opt/bitwarden_rs
+                    G_EXEC_OUTPUT=1 G_EXEC cargo build --features sqlite --release
+                    G_EXEC cd /tmp/$G_PROGRAM_NAME
+
+                    # Uninstall rust after compiling
+                    G_EXEC rustup self uninstall -y
```

Comment:
This will uninstall Rust from `/root/.cargo` right? What do you think about installing it to `/tmp`?
Or keeping Rust/cargo allow easy updates or has at an internal updater or otherwise requires a manual source download anyway (=> `dietpi-software reinstall 183` as easy solution)?

Reply:
It uses rust nightly, so it would probably have to re-download anyway (and yes it uninstalls from /root/.cargo). May be able to use /tmp, but I don't see any potential gains since it is removed afterwards anyway.

Reply:
It would reduce physical disk (SD card) writes and potentially speed up things a little... although CPU/compiling is what takes time here, not the disk reads/writes 😄.

Reply:
Oh yeah, /tmp isn't on the sd card, its a link to the ram.
I will try to see if I can change the install path

Reply:
There is no explicit option but the users home dir is used by default. A custom path can be force by setting the HOME variable, but the script then prints an ugly error message which as well cannot be muted via quite flag:
```
2020-11-20 18:03:24 root@VM-Buster:~# curl -sSfL https://sh.rustup.rs | HOME=/tmp dash -s -- -q -y --default-toolchain none
info: downloading installer
error: $HOME differs from euid-obtained home directory: you may be using sudo
error: $HOME directory: /tmp
error: euid-obtained home directory: /root
info: profile set to 'default'
info: default host triple is x86_64-unknown-linux-gnu
info: skipping toolchain installation


Rust is installed now. Great!

To get started you need Cargo's bin directory ($HOME/.cargo/bin) in your PATH
environment variable. Next time you log in this will be done
automatically.

To configure your current shell run source $HOME/.cargo/env
```
It could be wrapped into G_EXEC, with eval to allow the pipe:
```
2020-11-20 18:03:16 root@VM-Buster:~# G_EXEC eval 'curl -sSfL https://sh.rustup.rs | HOME=/tmp dash -s -- -y --default-toolchain none'
[  OK  ] eval curl -sSfL https://sh.rustup.rs | HOME=/tmp dash -s -- -y --default-toolchain none
```

Reply:
Wow, thanks for doing the research

Reply:
I have implemented it and tested installing and uninstalling on my machine. I have not tested the compiling, but it will probably also require the home override like the rustup uninstall

Reply:
I think `source "/tmp/.cargo/env"` covers the custom Rust dir, but does not hurt.